enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/messaging/SenderService.java

```diff
@@ -78,12 +75,17 @@ public void send( AdvertisedSocketAddress to, Message message )
                 return;
             }
 
-            channel( to ).send( message );
+            future = channel( to ).send( message );
         }
         finally
         {
             serviceLock.readLock().unlock();
         }
+
+        if ( block )
+        {
+            future.awaitUninterruptibly();
+        }
```

Comment:
I'm not familiar with this (or Netty `Future`'s) code so I'm missing context, but I'd normally expect an async service to return a `Future`, allowing the calling site to decide whether to chain callbacks, block, or just ignore.